### PR TITLE
TypeGraph: Create dummy containers

### PR DESCRIPTION
**Pull request by ajor**
These represent types which don't store any interesting data for us to measure, but which are required by a real container so can not be replaced with our own generated class types.

std::allocator often has bad DWARF, so it must be replaced after the DWARF is fixed up in Flattener. The others could be replaced earlier in the transformation process if desired, but I've left them all together for simplicity for now.

This fixes the folly::fbstring tests.

### CODE REVIEW

**Comment by ajor** [comment]
I also included a fix for the failing code coverage in this PR - see commit message for details

File: oi/type_graph/TypeIdentifier.cpp [hunk 1]
```diff
@@ -20,6 +20,15 @@
 
 namespace type_graph {
 
+// TODO:
+// - read these from a TOML file
```
Comment (JakeHillion) [inline oi/type_graph/TypeIdentifier.cpp:24]: can you create an issue for putting these in the config or creating container `.toml`s for them? I'd lean towards the config for simplicity but either works

File: oi/type_graph/TypeIdentifier.cpp [hunk 2]
```diff
@@ -20,6 +20,15 @@
...
+// - don't require specifying ctype and header
```
Comment (JakeHillion) [inline oi/type_graph/TypeIdentifier.cpp:25]: do you not need to specify the headers here? you could use "cstddef" in that case which is always included, but it looks to me like you do need the headers in weird edge cases

**Review by JakeHillion — APPROVED** [review]

**Comment by JakeHillion** [comment]
> ## [Codecov](https://app.codecov.io/gh/facebookexperimental/object-introspection/pull/185?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental) Report
> 
> > Merging [#185](https://app.codecov.io/gh/facebookexperimental/object-introspection/pull/185?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental) ([ea1c442](https://github.com/facebookexperimental/object-introspection/commit/ea1c442aa5d83fa9a105eef059d9277822a02ce2)) into [main](https://app.codecov.io/gh/facebookexperimental/object-introspection/commit/4dc9007166268f9c62d37083d8366ff6e766f466?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental) ([4dc9007](https://github.com/facebookexperimental/object-introspection/commit/4dc9007166268f9c62d37083d8366ff6e766f466)) will **increase** coverage by `63.82%`.
> > The diff coverage is `91.66%`.
> 
> ```diff
> @@            Coverage Diff             @@
> ##            main     #185       +/-   ##
> ==========================================
> + Coverage   0.48%   64.31%   +63.82%     
> ==========================================
>   Files         73       87       +14     
>   Lines       6817     9429     +2612     
>   Branches    1049     1553      +504     
> ==========================================
> + Hits          33     6064     +6031     
> + Misses      6777     2542     -4235     
> - Partials       7      823      +816     
> ```
> 
> [Impacted Files](https://app.codecov.io/gh/facebookexperimental/object-introspection/pull/185?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental) 	Coverage Δ 	
> [oi/type_graph/TypeIdentifier.h](https://app.codecov.io/gh/facebookexperimental/object-introspection/pull/185?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental#diff-b2kvdHlwZV9ncmFwaC9UeXBlSWRlbnRpZmllci5o) 	`100.00% <ø> (+100.00%)` 	⬆️
> [oi/ContainerInfo.cpp](https://app.codecov.io/gh/facebookexperimental/object-introspection/pull/185?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental#diff-b2kvQ29udGFpbmVySW5mby5jcHA=) 	`52.53% <50.00%> (+52.53%)` 	⬆️
> [oi/type_graph/TypeIdentifier.cpp](https://app.codecov.io/gh/facebookexperimental/object-introspection/pull/185?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental#diff-b2kvdHlwZV9ncmFwaC9UeXBlSWRlbnRpZmllci5jcHA=) 	`96.15% <100.00%> (+96.15%)` 	⬆️
> 
> ... and [79 files with indirect coverage changes](https://app.codecov.io/gh/facebookexperimental/object-introspection/pull/185/indirect-changes?src=pr&el=tree-more&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental)
> 
> 📣 We’re building smart automated test selection to slash your CI/CD build times. [Learn more](https://about.codecov.io/iterative-testing/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=facebookexperimental)

Is 91.66% accurate? I'm amazed!

File: oi/type_graph/TypeIdentifier.cpp (hunk 1 above)
Comment (ajor) [reply oi/type_graph/TypeIdentifier.cpp:24]: Agreed that just putting these in the config would be simpler - no need for a file for each

File: oi/type_graph/TypeIdentifier.cpp (hunk 2 above)
Comment (ajor) [reply oi/type_graph/TypeIdentifier.cpp:25]: Each of these dummy containers belongs to a real container and will only appear when that container is used. That means their header files will already be included by the real container.

Actually, typing this out has made me think that maybe "dummy container" should be a field in the container TOML of each real container that these belong to

Comment (ajor) [reply oi/type_graph/TypeIdentifier.cpp:25]: Oh, except std::allocator, which doesn't belong to any single container, but is used by all of them

**Comment by ajor** [comment]
> Is 91.66% accurate? I'm amazed!

We're at 100% coverage for the majority of type_graph files so I'd assume so!

**Pull request merged**